pkgs/by-name/ne/nexusmods-app/package.nix

```diff
@@ -59,8 +59,14 @@ buildDotnetModule (finalAttrs: {
   nugetDeps = ./deps.nix;
   mapNuGetDependencies = true;
 
-  dotnet-sdk = dotnetCorePackages.sdk_8_0;
-  dotnet-runtime = dotnetCorePackages.runtime_8_0;
+  # TODO: remove SDK 8; StrawberryShake currently needs it
+  dotnet-sdk =
+    with dotnetCorePackages;
+    combinePackages [
+      sdk_8_0
```

Comment:
this really should be documented or checked, but it'll use the `dotnet` CLI from the one at the head of the list, so generally it's a good idea to put the newest one first.

Also, have you tried with just runtime_8_0?  I'm curious if that work.

Reply:
Q: do both the sdk and the runtime get installed or is one only used for _building_ the derivation?

I initially went with sdk because I guessed that the SDK was used for building and the runtime was installed as a runtime dep to allow running.

I wanted to avoid inflating the final closure installed on end-users systems, but if you think a different approach is better that's fine by me :grinning:

Reply:
FYI @corngood I fixed the sdk ordering. As for your other question, see above :upside_down_face:

Reply:
I think @corngood's suggestion was to do:
```
combine packages [
  sdk_9_0
  runtime_8_0
]
``` so that if the tool only needs .NET 8.0 to run, we don't pull the whole SDK unnecessarily.

Reply:
> Q: do both the sdk and the runtime get installed or is one only used for building the derivation?

Usually the SDK is only used if you need to build a package or if your package is also a tool that's part of the SDK.

If you only need to run an already compiled program then use either the runtime or aspnetcore variants.

In this case, @corngood suggested that you try to use the runtime for 8.0 because if it's used for an external tool and not to build the application itself, it *might* not be needed, but the application needs .NET 9.0's SDK to be built so that one's a necessity.

Reply:
Thanks for clarifying. Seems to work as intended:

```nix
dotnet-sdk =
  with dotnetCorePackages;
  combinePackages [
    sdk_9_0
    runtime_8_0
  ];
```